### ISSUE TITLE
Differentiate between versions of a test with different InlineData

### ISSUE THREAD

**Issue by ryanbrandenburg** [dotnet/cli#3145]
## Steps to reproduce
1. Launch MusicStore in VS
2. Run the test SmoeTests_X64.WindowsOS (any of the InlineData versions will do)
3. Select the test after it finishes and see that it's run 9 times (the same number of time as the test has InlineData attributes)
## Expected  behavior

I would expect that each InlineData would only run itself and only run once.
## Actual behavior

Every InlineData runs each of the attributes, and will show as failed in Test Explorer if any of them failed.
## Environment data

`dotnet --info` output:
.NET Command Line Tools (1.0.0-preview2-002754)

Product Information:
 Version:            1.0.0-preview2-002754
 Commit SHA-1 hash:  93233fd496

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

**Comment by piotrpMSFT** [comment]
@ryanbrandenburg are you saying that dotnet-test-xunit executes tests inconsistently with other xunit runners, e.g. the xunit console runner?

/cc @livarcocc

**Comment by ryanbrandenburg** [comment]
After re-acquainting myself with this problem on returning from vacation I found that this is only true of `ConditionalTheory`'s, not of `Theory`'s in general.

And to answer your question @piotrpMSFT yes, this behavior doesn't show when running in the console, only in VS running `dotnet-test-xunit`.

Here's a quick test class that I whipped up while confirming everything:
```
public class ConditionalTheory
{
    private static int ConditionalTimesRun = 0;

    [ConditionalTheory]
    [InlineData(1)]
    [InlineData(2)]
    public void ConditionalTheoryRunOnce(int i)
    {
        ConditionalTimesRun++;

        if (ConditionalTimesRun > 2)
        {
            Assert.False(true, $"ConditionalTheory was run {ConditionalTimesRun} times");
        }
    }
}
```
It's my understanding that `dotnet-test-xunit` sends each individual test to xunit to be run rather than giving it a class like the xunit console runner does, which I would guess is where the problem lies, though it's not entirely clear to me why a `ConditionalTheory` would effect that where a normal `Theory` doesn't.
